helix-gateway/src/main/java/org/apache/helix/gateway/service/GatewayServiceManager.java

```diff
@@ -11,12 +11,23 @@
 import org.apache.helix.gateway.mock.MockProtoRequest;
 import org.apache.helix.gateway.mock.MockProtoResponse;
 
-public class ClusterManager {
+/**
+ * A top layer class that
+ *  1. get event from Handler
+ *  2. Maintain a gateway service registry, one gateway service maps to one Helix cluster
+ *  3. On init connect, create the participant manager
+ *  4. For ST reply message, update the tracker
+ */
+
+public class GatewayServiceManager {
```

Comment:
This manager should talk to the service instead of processor? The processor cannot proactively send message, right?